Sources/PointFree/Subscribe.swift

```diff
@@ -27,94 +28,139 @@ private let validateSubscribeData
     <<< validateCoupon
     <<< validateReferrer
 
-private func subscribe(_ conn: Conn<StatusLineOpen, Tuple3<User, SubscribeData, Referrer?>>)
-  -> IO<Conn<ResponseEnded, Data>> {
-
-    let referralDiscount: Cents<Int> = -18_00
+private func subscribe(
+  _ conn: Conn<StatusLineOpen, Tuple3<User, SubscribeData, Referrer?>>
+) -> IO<Conn<ResponseEnded, Data>> {
+
+  let (user, subscribeData, referrer) = lower(conn.data)
+  let referrerDiscount: Cents<Int> =
+    referrer?.stripeSubscription.discount?.coupon.id == Current.envVars.regionalDiscountCouponId
+      ? -9_00
+      : -18_00
+  let referredDiscount: Cents<Int> = subscribeData.useRegionalDiscount
+    ? -9_00
+    : -18_00
```

Comment:
this computes the referral discount we give out based on if the regional discount is applied.

this means the referred person only gets a $9 credit if they use the regional discount, and the referrer only gets $9 if they used the regional discount.